src/listener.coffee

```diff
@@ -39,7 +39,7 @@ class TextListener extends Listener
   # callback - A Function that is triggered if the incoming message matches.
   constructor: (@robot, @regex, @callback) ->
     @matcher = (message) =>
-      if message instanceof TextMessage
+      if message.match
```

Comment:
Wait. That's a lie. I was thinking of `matcher`. I still think we should do the test for `type == 'text'` though. Much more explicit of the intent (we want the `TextMessage` contract, not the 'I have a match method' contract)